File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
       int n=nums.size();
        vector<int>counts(n,0);
                
        for(int i=0; i<nums.size(); i++){
           for(int j=0; j<nums.size(); j++){
             if(nums[i] > nums[j]){
                 counts[i]++;
             }  
           } 
       } 
        return counts;
    }
};

File: 1365-how-many-numbers-are-smaller-than-the-current-number/1365-how-many-numbers-are-smaller-than-the-current-number.py
class Solution:
    def smallerNumbersThanCurrent(self, nums: List[int]) -> List[int]:
        length = len(nums)
        answer = []
        map_num = {}
        sorted_num = sorted(nums)
        for index, num in enumerate(sorted_num):
            if num not in map_num:
                map_num[num] = index
        
        for num in nums:
            smaller_count = map_num[num]
            answer.append(smaller_count)
            
        return answer